### PR TITLE
Fix#53/search error

### PR DESCRIPTION
**Pull request by q270311**

### CODE REVIEW

**Review by kepkaklaudia — APPROVED** [review]
Great! Good job 👍

File: src/common/Header/Search/index.js [hunk 1]
```diff
@@ -1,22 +1,35 @@
...
 import { useLocation } from "react-router";
+import { useState } from "react";
 
 const Search = () => {
-  const query = useQueryParameter(searchQueryParamName);
+  const queryURL = useQueryParameter(searchQueryParamName);
   const replaceQueryParameter = useReplaceQueryParameter();
   const location = useLocation();
+  const [query, setQuery] = useState(queryURL);
+  const [wait, setWait] = useState(true);
```
Comment (panijoanna) [inline src/common/Header/Search/index.js:17]: Perhaps it would be better to call the state isWait or isFetching instead wait?

**Review by panijoanna — APPROVED** [review]
I would change the name of the state to isWait/isFetching instead of wait :)

File: src/common/Header/Search/index.js (hunk 1 above)
Comment (q270311) [reply src/common/Header/Search/index.js:17]: Good idea, isWait sounds better :)

File: src/common/Header/Search/index.js [hunk 2]
```diff
@@ -1,22 +1,35 @@
...
+    if (wait) {
+      setWait(false);
+      setTimeout(() => {
+        replaceQueryParameter({
+          baseUrl: location.pathname.includes("movie") ? toMoviesList() : toPeopleList(),
+          key: searchQueryParamName,
+          value: target.value.trim() !== "" ? target.value : undefined,
+        });
+        setWait(true);
+      }, 3000);
```
[the remark this answers is not in the document]
Comment (q270311) [reply src/common/Header/Search/index.js:31]: I will reduce to 2 sec

**Pull request merged**